### DISCUSSION TITLE
用 data.table 做数据处理的笔记 | /project/data.table/

### DISCUSSION THREAD

**Discussion by giscus** [earfanfan/yuanfan#585] [Announcements]
https://yuanfan.rbind.io/project/data.table/

**Comment by yihui** [comment]
虽然我十多年前就知道 data.table 的 `:=` 操作符了，但看了博主的笔记才知道原来还有 `':='(x1 = value1, x2 = value2, ...)` 这种多列用法，以前我只知道操作单列的 `x := value` 用法。

**Reply by earfanfan** [reply]
这个包最早不是2016年才发布的么？`:=`和`[`其实非常灵活，但我写笔记的时候做了一些取舍，没有探索更多种写法。

话又说回来，我最近好像终于找到你说的猫屎果果了，学名可能叫做“白英”。我现在找到的实物，藤本，大小一样，表面也很光滑，但是现在它还是绿色的，我猜是还没长熟，哈哈哈，我每天早上遛狗都会经过它，会瞅一眼它撒时候开始变红。等它长熟了，我就摘一颗下来，捏爆它，这样的话，就会符合你所描述的所有特点了。

![0f497d6e820f818bb1557424c0d71f8](https://github.com/earfanfan/yuanfan/assets/29876335/29651d29-6a40-447d-8373-29445ab3f5c9)

**Reply by yihui** [reply]
> 这个包最早不是2016年才发布的么？

2006：https://cran.r-project.org/src/contrib/Archive/data.table/

> 我最近好像终于找到你说的猫屎果果了

这这这……这都被你找到了？！是它，是它，就是它！
